api_app/playbooks_manager/controller.py

```diff
@@ -0,0 +1,283 @@
+                        f"Analyzer {a_name} has a wrong queue."
+                        f"Setting to `{DEFAULT_QUEUE}`"
+                    )
+                    queue = DEFAULT_QUEUE
+                # get soft_time_limit
+                soft_time_limit = aa.config.soft_time_limit
+                task_signatures.append(
+                    tasks.run_analyzer.signature(
+                        args,
+                        {},
+                        queue=queue,
+                        soft_time_limit=soft_time_limit,
+                        task_id=task_id,
+                    )
+                )
+
+            except NotRunnableAnalyzer as e:
+                logger.warning(e)
+        
+        for c_name in connectors:
+            if c_name in connectors_used:
+                continue
+            cc = ConnectorConfig.get(c_name)
+            c_params = connectors.get(c_name)
+
+            try:
+                if cc is None:
+                    raise NotRunnableConnector(
+                        f"{c_name} won't run: not available in configuration"
+                    )
+                    
+                # if disabled or unconfigured (this check is bypassed in STAGE_CI)
+                if not cc.is_ready_to_use and not settings.STAGE_CI:
+                    raise NotRunnableConnector(
+                        f"{c_name} won't run: is disabled or unconfigured"
+                    )
+                    
+                connectors_used.append(c_name)
+                task_id = uuid()
+                args = [
+                    job_id,
+                    cc.asdict(),
+                    {"runtime_configuration": c_params, "task_id": task_id},
+                    p_name
+                ]
+
+                # get celery queue
+                queue = cc.config.queue
+                if queue not in settings.CELERY_QUEUES:
+                    logger.error(
+                        f"Connector {c_name} has a wrong queue."
+                        f" Setting to `{DEFAULT_QUEUE}`"
+                    )
+                queue = DEFAULT_QUEUE
+                # get soft_time_limit
+                soft_time_limit = cc.config.soft_time_limit
+                # create task signature and add to list
+                task_signatures.append(
+                    tasks.run_connector.signature(
+                        args,
+                        {},
+                        queue=queue,
+                        soft_time_limit=soft_time_limit,
+                        task_id=task_id,
+                    )
+                )
+
+            except NotRunnableConnector as e:
+                logger.warning(e)
+
+    job.update_analyzers_to_execute(analyzers_used)
+    job.update_connectors_to_execute(connectors_used)
+
+    runner = chord(task_signatures)
+    cb_signature = tasks.post_all_playbooks_finished.signature(
+        [job.pk], immutable=True
+    )
+    runner(cb_signature)
+    return None
```

Comment:
Why not writing a wrapper for `start_analyzers`? I think you wrote a lot of code, that I really don't think we need to maintain. Just do your setup, call `start_analyzers` and your finally your post_execution code. Probably i'm wrong and i'm not seeing the motivation for having all this

Reply:
I think I can add a `playbook_mode` parameter to `start_analyzers` to solve this issue because there is some extra logic that goes in there as well. Then I would be able to simply call it.